amgut/db/0042.sql

```diff
@@ -0,0 +1,129 @@
+COMMENT ON COLUMN ag."source".ag_login_id IS 'Points to the "user" that owns this source.';
+
+CREATE TABLE ag.source_barcodes_surveys (
+	source_id            uuid  NOT NULL,
+	barcode              varchar  ,
+	survey_id            varchar
+ );
+
+CREATE INDEX idx_source ON ag.source_barcodes_surveys ( barcode );
+CREATE INDEX idx_source_0 ON ag.source_barcodes_surveys ( survey_id );
+CREATE INDEX idx_source_barcodes_surveys ON ag.source_barcodes_surveys ( source_id );
+
+COMMENT ON COLUMN ag.source_barcodes_surveys.source_id IS 'Points to information about the source, like its name.';
+COMMENT ON COLUMN ag.source_barcodes_surveys.barcode IS 'Points to barcode(s) that are assigned to this source.';
+COMMENT ON COLUMN ag.source_barcodes_surveys.survey_id IS 'Points to survey(s) that are assigned to this source.';
+
+ALTER TABLE ag.source_barcodes_surveys ADD CONSTRAINT fk_source_barcode FOREIGN KEY ( barcode ) REFERENCES barcodes.barcode( barcode );
+ALTER TABLE ag.source_barcodes_surveys ADD CONSTRAINT fk_source_barcodes_surveys FOREIGN KEY ( source_id ) REFERENCES ag."source"( source_id );
+
+-- Re-use Daniels SQL statements to fill the new tables:
+
+-- create a view of all of the participants with multiple survey ids
+CREATE OR REPLACE TEMP VIEW multiple_ids AS
+SELECT ag_login_id,
+       participant_name,
+       unnest(survey_ids) AS survey_id
+FROM (SELECT DISTINCT ag_login_id,
+                      participant_name,
+                      array_agg(survey_id) AS survey_ids
+      FROM ag.ag_login_surveys
+      GROUP BY ag_login_id, participant_name) AS foo
+where array_length(survey_ids, 1) > 1;
+
+-- create a view of all of the barcodes associated with the participants
+CREATE OR REPLACE TEMP VIEW participant_barcodes AS
+SELECT ag_login_id, participant_name, barcode
+FROM multiple_ids mi
+JOIN ag.ag_kit_barcodes USING(survey_id);
+
+-- produce the cartesian product of barcodes and survey ids into a temp
+-- view that is in a common column structure to ag_kit_barcodes
+CREATE OR REPLACE TEMP VIEW like_ag_kit_barcodes AS
+SELECT ag_kit_barcode_id,
+       ag_kit_id,
+       barcode,
+       CASE WHEN multiple_ids.survey_id IS NOT NULL
+            THEN multiple_ids.survey_id
+            ELSE ag.ag_kit_barcodes.survey_id
+       END,
+       sample_barcode_file,
+       sample_barcode_file_md5,
+       site_sampled,
+       sample_date,
+       sample_time,
+       notes,
+       environment_sampled,
+       moldy,
+       overloaded,
+       other,
+       other_text,
+       date_of_last_email,
+       results_ready,
+       withdrawn,
+       refunded,
+       deposited,
+			 ag_login_id,
+			 participant_name
+FROM participant_barcodes
+JOIN multiple_ids USING (ag_login_id, participant_name)
+FULL JOIN ag.ag_kit_barcodes USING (barcode);
+
+-- enrich with ag_login_id and participant_name; the later becomes source_name
+CREATE OR REPLACE TEMP VIEW foo AS
```

Comment:
Please provide useful names to the tables (even if they're temp) so a reader can understand better the code.